docs/rfcs/batching-process-design.md

```diff
@@ -0,0 +1,282 @@
+  in this configuration.
+
+The queue sender does not consider the caller's context deadline when it attempts
+to enqueue the request. If the queue is full, the queue sender returns a queue-full
+error immediately.
+
+### Feature matrix
+
+| Support area | Batch processor | Batch sender | Explanation |
+| -----| -- |  -- | -- |
+| Merges requests | Yes | Yes | Does it merge smaller into larger batches? |
+| Splits requests | Yes | Yes, however sequential | Does it split larger into smaller batches? |
+| Cancellation  | No | No | Does it respect caller cancellation? |
+| Deadline  | No | No | Does it set an outgoing deadline? |
+| Metadata  | Yes | No | Can it batch by metadata key value(s)? |
+| Tracing  | No | No | Instrumented for tracing? |
+| Error transmission | No | Yes | Are export errors returned to callers? |
+| Concurrency enabled | No | Merge: Yes<br> MergeSplit: No | Does the component allow concurrent export? |
+| Independence | Yes | No | Are all data exported independently? |
+
+### Change proposal
+
+#### Queue sender defaults
+
+A central aspect of this presentation focuses on the queue sender,
+which along with one of the batching processors defines the behavior
+of most OpenTelemetry pipelines. Therefore, the default queue sender
+behavior is critical.
+
+This proposal argues in favor of the user, who does not want default
+behavior that leads to data loss. This requires one or more changes
+in the exporterhelper:
+
+1. Disable the queue sender by default. In this configuration, requests
+   become synchronous through the batch processor, and responses are delayed
+   until whole batches covering the caller's input have been processed.
+   Assuming the other requirements in this proposal are also carried out,
+   this means that pipelines will block each caller awaiting end-to-end
+   success, by default.
+2. Prevent start-up without a persistent queue configured; users would
+   have to opt-in to the in-memory queue if they want to return success
+   with no persistent store and not await end-to-end success.
+
+#### Batch processor: required
+
+The batch processor MUST be modified to achieve the following
+outcomes:
+
+- Allow concurrent exports. When the processor has a complete batch of
+  data available to send, it will send the data immediately.
+- Transmit errors back to callers. Callers will be blocked
+  while one or more requests are issued and wait on responses.
+- Respect context cancellation. Callers will return control
+  to the pipeline when their context is cancelled.
+
+#### Batch sender: required
+
+The batch sender MUST be modified to achieve the following
+outcomes:
+
+- Allow concurrent splitting. When multiple full-size requests
+  are produced from an input, they are exported concurrently and
+  independently.
+- Recognize partial splitting. When a request is split, leaving
+  part of a request that is not full, it remains in the active
+  request (i.e., still pending).
```

Comment:
I'm afraid this might introduce a significant delay when several incoming requests that are partially split block each other until the timeout is reached.